src/models/deployment-service.ts

```diff
@@ -0,0 +1,100 @@
+import Logger from "bunyan";
+import { envVars } from "config/env";
+import { getLogger } from "config/logger";
+import { dynamodb as ddb } from "models/dynamodb";
+
+const defaultLogger = getLogger("DeploymentDynamoLogger");
+
+const ONE_YEAR_IN_MILLISECONDS = 365 * 24 * 60 * 60 * 1000;
+
+export const saveDeploymentInfo = async (deploymentInfo : {
+	gitHubBaseUrl: string
+	gitHubInstallationId: number;
+	repositoryId: number;
+	commitSha: string;
+	description: string;
+	env: string;
+	status: string;
+	createdAt: Date;
+}, logger: Logger) => {
+	logger.debug("Saving deploymentInfo to db");
+	const result = await ddb.putItem({
+		TableName: envVars.DYNAMO_DEPLOYMENT_HISTORY_TABLE_NAME,
+		Item: {
+			Id: { "S": getKey(deploymentInfo) },
+			StatusCreatedAt: { "N": String(deploymentInfo.createdAt.getTime()) },
+			GitHubInstallationId: { "N": String(deploymentInfo.gitHubInstallationId) },
+			RepositoryId: { "N": String(deploymentInfo.repositoryId) },
+			CommitSha: { "S": deploymentInfo.commitSha },
+			Description: { "S": deploymentInfo.description },
+			Env: { "S": deploymentInfo.env },
+			Status: { "S": deploymentInfo.status },
+			ExpiredAfter: { "N": String(Math.floor((deploymentInfo.createdAt.getTime() + ONE_YEAR_IN_MILLISECONDS) / 1000)) }
```

Comment:
What about leap years? kidding

Reply:
We forget things in leap years, so is TTL.